adapters/AppLovin/AppLovinAdapter/GADMAdapterAppLovin.m

```diff
@@ -193,15 +184,195 @@ - (ALInterstitialAd *)interstitial {
   return _interstitial;
 }
 
-- (void)log:(NSString *)format, ... {
-  if (!loggingEnabled) return;
+#pragma mark - Banner Implementation
+
+- (void)getBannerWithSize:(GADAdSize)adSize
+{
+    [self log: @"Requesting AppLovin banner of size %@", NSStringFromGADAdSize(adSize)];
+    
+    // Convert requested size to AppLovin Ad Size
+    ALAdSize *appLovinAdSize = [self appLovinAdSizeFromRequestedSize: adSize];
+    if ( appLovinAdSize )
+    {
+        CGSize size = CGSizeFromGADAdSize(adSize);
+        
+        self.adView = [[ALAdView alloc] initWithFrame: CGRectMake(0.0f, 0.0f, size.width, size.height)
+                                                 size: appLovinAdSize
+                                                  sdk: self.sdk];
+        
+        AppLovinAdMobBannerDelegate *delegate = [[AppLovinAdMobBannerDelegate alloc] initWithParentAdapter: self];
+        self.adView.adLoadDelegate = delegate;
+        self.adView.adDisplayDelegate = delegate;
+        
+        if ( [self.adView respondsToSelector: @selector(setAdEventDelegate:)] )
+        {
+            self.adView.adEventDelegate = delegate;
+        }
+        
+        [self.sdk.adService loadNextAd: appLovinAdSize andNotify: delegate];
+    }
+    else
+    {
+        [self log: @"Failed to create an AppLovin Banner with invalid size"];
+        
+        NSError *error = [NSError errorWithDomain: kGADMAdapterAppLovinErrorDomain
+                                             code: kGADErrorMediationInvalidAdSize
+                                         userInfo: nil];
+        
+        [self.connector adapter: self didFailAd: error];
+    }
+}
+
+- (BOOL)isBannerAnimationOK:(GADMBannerAnimationType)animType
+{
+    return YES;
+}
+
+#pragma mark - Banner Utility Methods
+
+- (ALAdSize *)appLovinAdSizeFromRequestedSize:(GADAdSize)size
+{
+    if ( GADAdSizeEqualToSize(kGADAdSizeBanner, size ) || GADAdSizeEqualToSize(kGADAdSizeLargeBanner, size ) )
+    {
+        return [ALAdSize sizeBanner];
+    }
+    else if ( GADAdSizeEqualToSize(kGADAdSizeMediumRectangle, size) )
+    {
+        return [ALAdSize sizeMRec];
+    }
+    else if ( GADAdSizeEqualToSize(kGADAdSizeLeaderboard, size) )
+    {
+        return [ALAdSize sizeLeader];
+    }
+    // This is not a concrete size, so attempt to check for fluid size
+    else
```

Comment:
AppLovin's banners are actually have fluid widths by default. We specify 320x50 in our documentation because that's the "common" size, but if a publisher sets the width to something else, the web content supporting our ads are smart enough to stretch horizontally. Would that work?

Reply:
Thank you for the clarification. That will work